File: apps/android/core/model/src/main/kotlin/app/gauja/core/model/status/DiscoverSliderType.kt
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
package app.gauja.core.model.status

// Seerr server/constants/discover.ts at 69f73a6f1486fdb51b8ddae9a94a8dfb629f461c.
sealed class DiscoverSliderType(val rawValue: Int?) {
    data object RECENTLY_ADDED : DiscoverSliderType(1)

    data object RECENT_REQUESTS : DiscoverSliderType(2)

    data object PLEX_WATCHLIST : DiscoverSliderType(3)

    data object TRENDING : DiscoverSliderType(4)

    data object POPULAR_MOVIES : DiscoverSliderType(5)

    data object MOVIE_GENRES : DiscoverSliderType(6)

    data object UPCOMING_MOVIES : DiscoverSliderType(7)

    data object STUDIOS : DiscoverSliderType(8)

    data object POPULAR_TV : DiscoverSliderType(9)

    data object TV_GENRES : DiscoverSliderType(10)

    data object UPCOMING_TV : DiscoverSliderType(11)

    data object NETWORKS : DiscoverSliderType(12)

    data object TMDB_MOVIE_KEYWORD : DiscoverSliderType(13)

    data object TMDB_MOVIE_GENRE : DiscoverSliderType(14)

    data object TMDB_TV_KEYWORD : DiscoverSliderType(15)

    data object TMDB_TV_GENRE : DiscoverSliderType(16)

    data object TMDB_SEARCH : DiscoverSliderType(17)

    data object TMDB_STUDIO : DiscoverSliderType(18)

    data object TMDB_NETWORK : DiscoverSliderType(19)

    data object TMDB_MOVIE_STREAMING_SERVICES : DiscoverSliderType(20)

    data object TMDB_TV_STREAMING_SERVICES : DiscoverSliderType(21)

    data class Unrecognized(val value: Int?) : DiscoverSliderType(value)

    companion object {
        fun fromWire(raw: Int?): DiscoverSliderType =
            when (raw) {
                1 -> RECENTLY_ADDED
                2 -> RECENT_REQUESTS
                3 -> PLEX_WATCHLIST
                4 -> TRENDING
                5 -> POPULAR_MOVIES
                6 -> MOVIE_GENRES
                7 -> UPCOMING_MOVIES
                8 -> STUDIOS
                9 -> POPULAR_TV
                10 -> TV_GENRES
                11 -> UPCOMING_TV
                12 -> NETWORKS
                13 -> TMDB_MOVIE_KEYWORD
                14 -> TMDB_MOVIE_GENRE
                15 -> TMDB_TV_KEYWORD
                16 -> TMDB_TV_GENRE
                17 -> TMDB_SEARCH
                18 -> TMDB_STUDIO
                19 -> TMDB_NETWORK
                20 -> TMDB_MOVIE_STREAMING_SERVICES
                21 -> TMDB_TV_STREAMING_SERVICES
                else -> Unrecognized(raw)
            }
    }
}

File: apps/ios/Packages/Model/Sources/Model/status/DiscoverSliderType.swift
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
// Seerr server/constants/discover.ts at 69f73a6f1486fdb51b8ddae9a94a8dfb629f461c.
public enum DiscoverSliderType: Sendable, Equatable, Codable {
    case recentlyAdded
    case recentRequests
    case plexWatchlist
    case trending
    case popularMovies
    case movieGenres
    case upcomingMovies
    case studios
    case popularTv
    case tvGenres
    case upcomingTv
    case networks
    case tmdbMovieKeyword
    case tmdbMovieGenre
    case tmdbTvKeyword
    case tmdbTvGenre
    case tmdbSearch
    case tmdbStudio
    case tmdbNetwork
    case tmdbMovieStreamingServices
    case tmdbTvStreamingServices
    case unrecognized(Int?)

    public init(rawValue: Int?) {
        switch rawValue {
        case 1: self = .recentlyAdded
        case 2: self = .recentRequests
        case 3: self = .plexWatchlist
        case 4: self = .trending
        case 5: self = .popularMovies
        case 6: self = .movieGenres
        case 7: self = .upcomingMovies
        case 8: self = .studios
        case 9: self = .popularTv
        case 10: self = .tvGenres
        case 11: self = .upcomingTv
        case 12: self = .networks
        case 13: self = .tmdbMovieKeyword
        case 14: self = .tmdbMovieGenre
        case 15: self = .tmdbTvKeyword
        case 16: self = .tmdbTvGenre
        case 17: self = .tmdbSearch
        case 18: self = .tmdbStudio
        case 19: self = .tmdbNetwork
        case 20: self = .tmdbMovieStreamingServices
        case 21: self = .tmdbTvStreamingServices
        default: self = .unrecognized(rawValue)
        }
    }

    public var rawValue: Int? {
        switch self {
        case .recentlyAdded: 1
        case .recentRequests: 2
        case .plexWatchlist: 3
        case .trending: 4
        case .popularMovies: 5
        case .movieGenres: 6
        case .upcomingMovies: 7
        case .studios: 8
        case .popularTv: 9
        case .tvGenres: 10
        case .upcomingTv: 11
        case .networks: 12
        case .tmdbMovieKeyword: 13
        case .tmdbMovieGenre: 14
        case .tmdbTvKeyword: 15
        case .tmdbTvGenre: 16
        case .tmdbSearch: 17
        case .tmdbStudio: 18
        case .tmdbNetwork: 19
        case .tmdbMovieStreamingServices: 20
        case .tmdbTvStreamingServices: 21
        case .unrecognized(let raw): raw
        }
    }
}
